### PR TITLE
Support Mock Compute TargetTcpProxy

### PR DESCRIPTION
**Pull request by gemmahou**
### Change description

based on #2628 

-Add mocks for global compute targettcpproxy
-Add tests case for computeforwardingrule with target TCP proxy
-Collect logs

### Tests you have done

- [ ] Run `make ready-pr` to ensure this PR is ready for review.
- [ ] Perform necessary E2E testing for changed resources.

### CODE REVIEW

File: pkg/test/resourcefixture/testdata/sensitivefield/sensitivevaluesimple/_generated_object_sensitivevaluesimple.golden.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  - cnrm.cloud.google.com/finalizer
+  - cnrm.cloud.google.com/deletion-defender
+  generation: 2
+  labels:
+    cnrm-test: "true"
+  name: computesslcertificate-${uniqueId}
+  namespace: ${uniqueId}
+spec:
+  certificate:
+    value: |
```
Comment (yuwenma) [inline pkg/test/resourcefixture/testdata/sensitivefield/sensitivevaluesimple/_generated_object_sensitivevaluesimple.golden.yaml:19]: Ops. You may want to change this to fake data.

File: pkg/test/resourcefixture/testdata/sensitivefield/sensitivevaluefromsecret/_generated_object_sensitivevaluefromsecret.golden.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  certificate:
+    valueFrom:
+      secretKeyRef:
+        key: certificate
+        name: secret-${uniqueId}
+  location: global
+  privateKey:
+    valueFrom:
+      secretKeyRef:
+        key: privateKey
```
Comment (yuwenma) [inline pkg/test/resourcefixture/testdata/sensitivefield/sensitivevaluefromsecret/_generated_object_sensitivevaluefromsecret.golden.yaml:27]: May I ask what is the field `Key` refers to?

File: pkg/test/resourcefixture/testdata/basic/sql/v1beta1/sqlinstance/sqlinstance-privatenetwork/_http.log [hunk 3]
```diff
@@ -2052,10 +2124,46 @@ X-Xss-Protection: 0
...
-  "status": "DONE"
+  "status": "RUNNING",
+  "targetId": "${addressesId}",
+  "targetLink": "https://www.googleapis.com/compute/v1/projects/${projectId}/global/addresses/computeaddress2-${uniqueId}",
+  "user": "user@example.com"
+}
+
+---
+
+GET https://compute.googleapis.com/compute/v1/projects/${projectId}/global/operations/${operationID}?alt=json&prettyPrint=false
```
Comment (yuwenma) [inline pkg/test/resourcefixture/testdata/basic/sql/v1beta1/sqlinstance/sqlinstance-privatenetwork/_http.log:2139]: qq: Is this from real GCP? I assume this should be `global/addresses`

**Review by yuwenma — COMMENTED** [review]
/lgtm

Overall looks good. One comment that we need to fix before merging this PR.

File: pkg/test/resourcefixture/testdata/sensitivefield/sensitivevaluesimple/_generated_object_sensitivevaluesimple.golden.yaml (hunk 1 above)
Comment (gemmahou) [reply pkg/test/resourcefixture/testdata/sensitivefield/sensitivevaluesimple/_generated_object_sensitivevaluesimple.golden.yaml:19]: This is the test certificate in [create.yaml](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/f6ffa102c61d86b92572951aae7f385e89994569/pkg/test/resourcefixture/testdata/sensitivefield/sensitivevaluesimple/create.yaml#L22)

File: pkg/test/resourcefixture/testdata/sensitivefield/sensitivevaluefromsecret/_generated_object_sensitivevaluefromsecret.golden.yaml (hunk 2 above)
Comment (gemmahou) [reply pkg/test/resourcefixture/testdata/sensitivefield/sensitivevaluefromsecret/_generated_object_sensitivevaluefromsecret.golden.yaml:27]: It's defined in the test [create.yaml](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/f6ffa102c61d86b92572951aae7f385e89994569/pkg/test/resourcefixture/testdata/sensitivefield/sensitivevaluefromsecret/create.yaml#L32) as well.

File: pkg/controller/direct/compute/forwardingrule_controller.go [hunk 4]
```diff
@@ -161,6 +161,16 @@ func (m *forwardingRuleModel) AdapterForObject(ctx context.Context, reader clien
 			}
 			obj.Spec.Target.TargetSSLProxyRef.External = targetSSLProxyRef.External
 		}
+
+		// Get target TCPProxy
```
Comment (yuwenma) [inline pkg/controller/direct/compute/forwardingrule_controller.go:165]: Does it mean this is an optional field and if not provided the compute service will assign a default? Can we add a test suite for this? no blocker, can be done in follow up PR

File: mockgcp/mockcompute/globaltargettcpproxyv1.go [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+	obj.ProxyHeader = req.GetTargetTcpProxiesSetProxyHeaderRequestResource().ProxyHeader
+	if err := s.storage.Update(ctx, fqn, obj); err != nil {
+		return nil, err
+	}
+
+	op := &pb.Operation{
+		TargetId:      obj.Id,
+		TargetLink:    obj.SelfLink,
+		OperationType: PtrTo("setProxyHeader"),
+		User:          PtrTo("user@example.com"),
```
Comment (yuwenma) [inline mockgcp/mockcompute/globaltargettcpproxyv1.go:130]: qq: Can user configure this field? Or it is always user@example.com?

File: pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computeforwardingrule/globalcomputeforwardingruletcp/create.yaml
Comment (yuwenma) [inline pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computeforwardingrule/globalcomputeforwardingruletcp/create.yaml:1]: Can we have both full and basic test suites? I cannot check the required/optional, mutable/immutable fields as well as an entirely new golden log in a single PR.

**Review by yuwenma — COMMENTED** [review]
/lgtm
/approve

Looks good! Let's split the test to basic and full test suites in follow up PRs.

**Pull request merged**